HexView.Wpf/ExtensionMethods.cs

```diff
@@ -16,8 +19,18 @@ public static class ExtensionMethods
         public static short ReadInt16BE(this BinaryReader reader)
         {
             var data = reader.ReadBytes(2);
-            Array.Reverse(data);
-            return BitConverter.ToInt16(data, 0);
+            return BinaryPrimitives.ReadInt16BigEndian(data);
+        }
+
+        /// <summary>
+        /// Reads a 2-byte signed integer in little-endian order and advances the current position of the stream by 2 bytes.
+        /// </summary>
+        /// <param name="reader">in-use BinaryReader.</param>
+        /// <returns>Int16 (signed short).</returns>
+        public static short ReadInt16LE(this BinaryReader reader)
+        {
+            var data = reader.ReadBytes(2);
+            return BinaryPrimitives.ReadInt16LittleEndian(data);
         }
```

Comment:
Thanks @Hogflash. I'm curious in #9 you mentioned:

> Filip, I spent some time looking at the endian issue, and after some investigation and testing, I updated the extension methods to use your shift-or extraction methods, which are about 2.5x faster than using the array.reverse or binaryPrimitives code. I added a test program used for checking performance of the endian extraction.

However the changes here appear to be using `BinaryPrimitives`. Was this the intent give the shift-or appears to be faster?

Reply:
Dang, I updated the wrong Extension Methods class - Based on your comments, let me remove this PR and provide the correct methods which are in the endian tester project.  I will also remove the tester from the commit code, as it just confuses things.  

On measuring the performance, I know the unused local variable statement is removed in the final compiled code, but my understanding is that in debug mode it is still present and gives some indication of the performance.

Thanks for your patience,  Gary

Reply:
> On measuring the performance, I know the unused local variable statement is removed in the final compiled code, but my understanding is that in debug mode it is still present and gives some indication of the performance.

In debug mode you're right that it won't get removed. Your performance measurements stand correct in debug mode, but I'm unsure if you would get the same relative difference in performance once the JIT compiler kicks in.

> --Hmm seems I don't know how to remove this PR. Is this something you can help me with, or can you do this as the project owner? I think I can cleanup my fork so that only the relevant two files are modified. Or, start over?

Typically you don't need to start over. You can modify your code on your current branch, add it to a new commit, then do an interactive rebase to squash the two commits back into one. You then force push to your Pull Request (PR) branch and the same changes will then be reflected here on GitHub. Good luck!